lib/Arch/PPC_C.cpp

```diff
@@ -0,0 +1,332 @@
+  explicit PPC_C(const remill::Arch *arch);
+  virtual ~PPC_C() = default;
+
+  llvm::Error AllocateSignature(FunctionDecl &fdecl,
+                                llvm::Function &func) override;
+
+ private:
+  llvm::Error BindParameters(llvm::Function &function, bool injected_sret,
+                             std::vector<ParameterDecl> &param_decls);
+
+  llvm::Error BindReturnValues(llvm::Function &function, bool &injected_sret,
+                               std::vector<ValueDecl> &ret_decls);
+
+  const std::vector<RegisterConstraint> &parameter_register_constraints;
+  const std::vector<RegisterConstraint> &return_register_constraints;
+};
+
+std::unique_ptr<CallingConvention>
+CallingConvention::CreatePPC_C(const remill::Arch *arch) {
+  return std::make_unique<PPC_C>(arch);
+}
+
+PPC_C::PPC_C(const remill::Arch *arch)
+    : CallingConvention(llvm::CallingConv::C, arch),
+      parameter_register_constraints(kParamRegConstraints),
+      return_register_constraints(kReturnRegConstraints) {}
+
+llvm::Error PPC_C::AllocateSignature(FunctionDecl &fdecl,
+                                     llvm::Function &func) {
+
+  // Bind return values first to see if we have injected an sret into the
+  // parameter list. Then, bind the parameters. It is important that we bind the
+  // return values before the parameters in case we inject an sret.
+  bool injected_sret = false;
+  auto err = BindReturnValues(func, injected_sret, fdecl.returns);
+  if (remill::IsError(err)) {
+    return err;
+  }
+
+  err = BindParameters(func, injected_sret, fdecl.params);
+  if (remill::IsError(err)) {
+    return err;
+  }
+
+  fdecl.return_stack_pointer_offset = 0;
+  fdecl.return_stack_pointer = arch->RegisterByName("R1");
+
+  fdecl.return_address.reg = arch->RegisterByName("LR");
+  fdecl.return_address.type = fdecl.return_address.reg->type;
+
+  return llvm::Error::success();
+}
+
+llvm::Error
+PPC_C::BindReturnValues(llvm::Function &function, bool &injected_sret,
+                        std::vector<anvill::ValueDecl> &ret_values) {
+
+  llvm::Type *ret_type = function.getReturnType();
+  LOG(INFO) << "Binding on return " << remill::LLVMThingToString(ret_type);
+  injected_sret = false;
+
+  // If there is an sret parameter then it is a special case.
+  if (function.hasStructRetAttr()) {
+    auto &value_declaration = ret_values.emplace_back();
+
+    value_declaration.type = llvm::PointerType::get(function.getContext(), 0);
+
+    if (!ret_type->isVoidTy()) {
+      return llvm::createStringError(
+          std::errc::invalid_argument,
+          "Function '%s' with sret-attributed parameter has non-void return type '%s'",
+          function.getName().str().c_str(),
+          remill::LLVMThingToString(ret_type).c_str());
+    }
+
+    // NOTE(alex): Not sure about this, check later.
+    //
+    // Indirect return values are passed by pointer through `R3`.
+    value_declaration.reg = arch->RegisterByName("R3");
+    return llvm::Error::success();
```

Comment:
Shouldn't this set `injected_sret` to `true`?

Reply:
[5e99152](https://github.com/lifting-bits/anvill/pull/340/commits/5e99152bc1fbc258ae6dcca43e0a96186137640b)

That variable is removed so no longer applies.